### DISCUSSION TITLE
Azure DevOps Git Repository tags within presets aren't supported

### DISCUSSION THREAD

**Discussion by rzsita** [renovatebot/renovate#29846] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Platform: Azure 
Renovate versions: 37.417.0

### Please tell us more about your question or problem

I'm trying to add presets to my Renovate config, but Renovate is using the config from the main branch regardless of defining the tag `v0.2.0`. Using `LOG_LEVEL=trace`, I observed that Renovate is unable to retrieve the file by tag and seems to fall back to the default branch instead.

#### renovate.json5
```json5
{
    $schema: 'https://docs.renovatebot.com/renovate-schema.json',
    extends: [
        'local>SharedInfra/renovate-config#v0.2.0',
    ],
    baseBranches: [
        'dev',
    ],
}
```

#### Message from Trace
```plaintext
TRACE: Found preset local>SharedInfra/renovate-config#v0.2.0
       "presetConfig": {
         "$id": "1",
         "innerException": null,
         "message": "TF401175: The version descriptor <Branch: v0.2.0> could not be resolved to a version in the repository renovate-config",
         "typeName": "Microsoft.TeamFoundation.Git.Server.GitUnresolvableToCommitException, Microsoft.TeamFoundation.Git.Server",
         "typeKey": "GitUnresolvableToCommitException",
         "errorCode": 0,
         "eventId": 3000
       }
```

#### Quick look at the Microsoft API documentation
Based on my review of the underlying Microsoft API documentation, it appears that a `versionType` must be specified to select tags instead of branches. However, this `versionType` is not currently set in the presets, preventing the selection of tags.

For more details, refer to the Microsoft documentation: [Azure DevOps REST API - Get Items](https://learn.microsoft.com/en-us/rest/api/azure/devops/git/items/get?view=azure-devops-rest-7.1&tabs=HTTP)

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by viceice** [comment]
how should we know that it's a tag instead of a branch?

**Reply by rzsita** [reply]
Thank you for responding so quickly.

Good question! In my opinion, there are several approaches to achieve this:

1. **Align the behavior with the document (tag, not branch):**
   ![image](https://github.com/renovatebot/renovate/assets/46534804/8d50ddbb-bbcb-46fd-b429-4d2272063fa2)
   This approach means there is no branch support.

2. **Prefix the name with GT (Git tag) or GB (Git branch):**
   - `local>SharedInfra/renovate-config#GTv0.2.0`
   - `local>SharedInfra/renovate-config#GBdev`
   - If there is no GT or GB prefix, use the tag by default:
     `local>SharedInfra/renovate-config#v0.2.0`

3. **Use refs/tags/ or refs/heads/:**
   - `local>SharedInfra/renovate-config#refs/tags/v0.2.0`
   - `local>SharedInfra/renovate-config#refs/heads/dev`
   - If there is no refs/tags/ or refs/heads prefix, use the tag by default:
     `local>SharedInfra/renovate-config#v0.2.0`

**Reply by rzsita** [reply]
@viceice What are your thoughts on my proposal?
